Fixed some miscellaneous deprecations

LumioseSil opened the pull request:
#### b9f7d5033a298c6b3dbec8c7d288ea19e13dc067
<pre>
Fixed some miscellaneous deprecations
<a href="https://bugs.webkit.org/show_bug.cgi?id=251865">https://bugs.webkit.org/show_bug.cgi?id=251865</a>

Reviewed by NOBODY (OOPS!).

Just some low-hanging fruit to deal with.
</pre><!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/4d4a6114e9c3fddbb07e2b1a0ffcdc58f309e0c7

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/108573 "Failed to checkout and rebase branch from PR 9765") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/17671 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/41424 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/117680 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/12/builds/117878 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/19122 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/8948 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/100802 "Built successfully") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/114340 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/14332 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/97559 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/35/builds/42300 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/96309 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/29200 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/84013 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/10481 "Built successfully") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/70/builds/30549 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/11237 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/84/builds/7463 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/16628 "Built successfully") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/62/builds/50144 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/12826 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

emw-apple commented on Source/WebKitLegacy/mac/WebView/WebView.mm:
```diff
@@ -2312,10 +2314,9 @@ - (void)_dispatchTileDidDraw:(CALayer*)tile
-    ALLOW_DEPRECATED_DECLARATIONS_BEGIN
-    if (!OSAtomicCompareAndSwap32(0, 1, &_private->didDrawTiles))
+    BOOL old = NO;
+    if (!atomic_compare_exchange_strong((_Atomic BOOL*)&_private->didDrawTiles, &old, YES))
```
Not my area of expertise, but I think we'd prefer using an atomic operation from `<wtf/Atomics.h>`. I see that this would be the first non-3rd-party use of `atomic_compare_exchange_strong` in the codebase.

emw-apple commented on Source/WebKit/Configurations/WebKit.xcconfig:
```diff
@@ -190,9 +190,9 @@ OTHER_LDFLAGS = $(inherited) -iframework"$(SDK_DIR)$(SYSTEM_LIBRARY_DIR)/Private
 
-// FIXME: Remove -reexport_install_name once rdar://problem/30820233 is fixed.
+// FIXME: Remove -reexport_framework once rdar://problem/30820233 is fixed.
 OTHER_TAPI_FLAGS = $(OTHER_TAPI_FLAGS_$(WK_COCOA_TOUCH));
-OTHER_TAPI_FLAGS_cocoatouch = $(inherited) -reexport_install_name $(WK_ALTERNATE_WEBKIT_SDK_PATH)$(SYSTEM_LIBRARY_DIR)/PrivateFrameworks/WebKitLegacy.framework/WebKitLegacy -extra-private-header $(SRCROOT)/Platform/ExtraPrivateSymbolsForTAPI.h -extra-public-header $(SRCROOT)/Platform/ExtraPublicSymbolsForTAPI.h -exclude-private-header $(BUILT_PRODUCTS_DIR)/$(PRIVATE_HEADERS_FOLDER_PATH)/NSURLDown ...
+OTHER_TAPI_FLAGS_cocoatouch = $(inherited) -reexport_framework $(WK_ALTERNATE_WEBKIT_SDK_PATH)$(SYSTEM_LIBRARY_DIR)/PrivateFrameworks/WebKitLegacy.framework/WebKitLegacy -extra-private-header $(SRCROOT)/Platform/ExtraPrivateSymbolsForTAPI.h -extra-public-header $(SRCROOT)/Platform/ExtraPublicSymbolsForTAPI.h -exclude-private-header $(BUILT_PRODUCTS_DIR)/$(PRIVATE_HEADERS_FOLDER_PATH)/NSURLDownloa ...
```
The ideal cleanup here would be to clear the FIXME and delete the `-reexport` argument altogether. You can see from open-source build logs that the radar has been fixed. For example, in the build log for https://build.webkit.org/#/builders/661/builds/4787, we ran:

```
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/tapi installapi 
-verify-against /Volumes/Data/worker/Apple-iOS-16-Release-Build/build/WebKitBuild/Release-iphoneos/WebKit.framework/WebKit 
…
-reexport_framework WebKitLegacy 
…
-reexport_install_name /System/Library/PrivateFrameworks/WebKitLegacy.framework/WebKitLegacy 
-extra-private-header…
```

So the build system is passing `-reexport_framework` for us, and we can remove our manual workaround.

AZero13 replied:
Tried that and it resulted in build errors. Will try again though.

emw-apple replied:
If that doesn't work, I recommend we do not change this line. As far as I can tell, `-reexport_install_name` isn't deprecated (could you show me if I'm wrong?). And `-reexport_framework` takes a framework name, not a path:

```
> xcrun tapi installapi -help
…
  -reexport_framework <value>
                          Re-export the specified framework
```

Ultimately, this feels like something that isn't broken or deprecated, and shouldn't be changed, until we can resolve the FIXME.

emw-apple replied:
Marking as unresolved, please address before merging.

aproskuryakov replied:
This FIXME is about using `REEXPORTED_FRAMEWORK_NAMES` instead of `OTHER_TAPI_FLAGS` for this. But I'm not sure if that's equivalent, Apple engineers can consult rdar://61788689 for more details.

emw-apple commented on Source/WebKitLegacy/mac/WebView/WebView.mm:
```diff
@@ -2312,10 +2312,8 @@ - (void)_dispatchTileDidDraw:(CALayer*)tile
 
-    ALLOW_DEPRECATED_DECLARATIONS_BEGIN
-    if (!OSAtomicCompareAndSwap32(0, 1, &_private->didDrawTiles))
+    if (WTF::atomicCompareExchangeWeakRelaxed(&_private->didDrawTiles, NO, YES))
```
Missing a negation:
```suggestion
    if (!WTF::atomicCompareExchangeWeakRelaxed(&_private->didDrawTiles, NO, YES))
```

emw-apple commented on Source/WebKitLegacy/mac/WebView/WebHTMLView.mm:
```diff
@@ -6153,10 +6153,7 @@ - (void)attachRootLayer:(CALayer *)layer
-        CGPoint layerPosition = NSPointToCGPoint([self convertPointToBase:[_private->layerHostingView frame].origin]);
-        ALLOW_DEPRECATED_DECLARATIONS_END
-        [viewLayer setPosition:layerPosition];
+        [viewLayer setPosition:NSPointToCGPoint([self convertPointToBacking:[_private->layerHostingView frame].origin])];
     }
```
Have you tested this? I'm not an AppKit expert, but it's not clear to me from API documentation that this is a change that's safe to make mechanically.

AZero13 replied:
I absolutely have.

emw-apple replied:
Marking unresolved for sign-off from a reviewer with more AppKit knowledge than me.

pxlcoder replied:
It's not apparent that this change is safe to make. `convertPointToBase` and `convertPointToBacking` do not have exactly equal implementations.

I would also err on the side of caution when removing code in WebKitLegacy, since all code here is effectively deprecated.

@smfr do you recall why resolution independence is needed here? I wonder if this was actually intended to be in window coordinates?

smfr replied:
I do not recall. But I would prefer this PR be broken into one for each part.

LumioseSil committed:
Fixed some miscellaneous deprecations
https://bugs.webkit.org/show_bug.cgi?id=251865

Reviewed by NOBODY (OOPS!).

Just some low-hanging fruit to deal with.

emw-apple reviewed (changes requested):
I don't think this is ready yet! Marking some comments as unresolved that I think still require changes.

Just a policy note, please don't remove requested reviewers just because one reviewer has left feedback. WebKit is a huge project and we depend on having domain-specific reviewers to ensure quality. I'm re-requesting review from the people @nt1m added, who can provide additional sign-off.

That said, thank you for the ongoing contributions. All cleanup help is appreciated!

whsieh commented on Source/WebKit/UIProcess/ios/WKActionSheet.mm:
```diff
@@ -216,15 +216,11 @@ - (void)updateSheetPosition
     // - The completion of the view controller dismissal in willRotate.
     // (We cannot present something again until the dismissal is done)
 
-    BOOL isBeingPresented = [presentedViewController presentingViewController] || [self presentingViewController];
```
I'm pretty sure this is a separate local variable on purpose, to clarify the intended purpose of this check without requiring a comment. Let's either keep it as-is, or make a static helper.

AZero13 replied:
Not true, the comment above says "we cannot present something again until the dismissal is done" which means as long as the presentingViewController is active, nothing can be done.
